src/core/core.datasetController.js

```diff
@@ -287,6 +374,99 @@ helpers.extend(DatasetController.prototype, {
 		});
 	},
 
+	/**
+	 * @private
+	 */
+	_storeParsedData: storeParsedData,
+
+	/**
+	 * @private
+	 */
+	_parse: function(start, count) {
+		var me = this;
+		var meta = me.getMeta();
+		var data = me._data;
+		var i, ilen, item;
+
+		if (!data || data.length === 0) {
+			return;
+		}
+
+		for (i = start, ilen = start + count; i < ilen; ++i) {
+			item = data[i];
+			if (helpers.isArray(item)) {
+				me._parseArrayData(item, meta, i);
+			} else if (helpers.isObject(item)) {
+				me._parseObjectData(item, meta, i);
+			} else {
+				me._parsePlainData(item, meta, i);
+			}
+		}
+	},
+
+	/**
+	 * @private
+	 */
+	_parsePlainData: function(value, meta, index) {
+		var iScale = this._getIndexScale();
+		var vScale = this._getValueScale();
+		var labels = iScale._getLabels() || [];
+		var v = vScale._parse(value, index);
+		var iv = iScale !== vScale && index < labels.length
+			? iScale._parse(labels[index], index)
+			: index;
+		storeParsedData(meta, index, iScale, vScale, iv, v);
+	},
+
+	/**
+	 * @private
+	 */
+	_parseArrayData: function(arr, meta, index) {
+		var xScale = this.getScaleForId(meta.xAxisID);
+		var yScale = this.getScaleForId(meta.yAxisID);
+		var x = xScale._parse(arr[0], index);
+		var y = yScale._parse(arr[1], index);
+		storeParsedData(meta, index, xScale, yScale, x, y);
+	},
+
+	/**
+	 * @private
+	 */
+	_parseCustomObjectData: helpers.noop,
+
+	/**
+	 * @private
+	 */
+	_parseObjectData: function(obj, meta, index) {
+		var xScale = this.getScaleForId(meta.xAxisID);
+		var yScale = this.getScaleForId(meta.yAxisID);
+		var x = xScale._parseObject(obj, 'x', index);
+		var y = yScale._parseObject(obj, 'y', index);
+		var c = this._parseCustomObjectData(obj);
+		storeParsedData(meta, index, xScale, yScale, x, y, c);
+	},
+
+	/**
+	 * @private
+	 */
+	_getParsedValue: function(index, scale) {
+		return loadParsedData(this.getMeta(), index, scale.id);
```

Comment:
There's a bunch of places in this class that call `this.getMeta()`. Can we cache it in `_update` so that we don't have to make as many function calls?

Reply:
done